crates/proto/src/rr/domain/name.rs

```diff
@@ -56,6 +56,31 @@ impl Name {
         Ok(())
     }
 
+    /// Randomize the case of ASCII alpha characters in a name
+    pub fn randomize_label_case(&mut self) {
+        use rand::distributions::{Distribution, Standard};
+        let mut rand = rand::thread_rng();
+
+        // 16 bits should be enough to cover *most* queries without making another call to the RNG.
+        // Using u16 here resulted in repeated cases of the most-significant bit always being zero
+        let mut rand_bits: i16 = 0;
+
+        for (i, ref mut b) in self.label_data.iter_mut().enumerate() {
```

Comment:
Do we really need the `ref mut` keywords?

Reply:
To change the label bytes without array indexes, yes.  Changing this loop over to enumerate but still mutating the byte via an array index seems a) less concise and b) less clear in that the ascii alpha test would use the reference byte returned by enumerate but the mutation would use an array index on the self.label_data.

Reply:
Sorry, I like the idea of mutating through the binding, but I don't think we need `ref mut` specifically? This seems to compile fine:

```rust
        let mut foo = [b'f', b'o', b'o'];
        for (_, b) in foo.iter_mut().enumerate() {
            *b = *b ^ 0x20;
        }
```

Reply:
Oh, I understand.  I was getting an error earlier; now it seems to work fine though.